src/BatteryStateDisplayService.cpp

```diff
@@ -51,10 +51,20 @@ BatteryStateDisplayService::~BatteryStateDisplayService()
 void BatteryStateDisplayService::handleBatteryDataReceived(const BatteryData& batteryData)
 {
     batteryStateOfChargeService_.addData(batteryData);
-
-    QTextStream(stdout) << "Voltage: " << batteryData.voltage
+    batteryStateOfChargeService_.isCharging();
+    if (batteryStateOfChargeService_.isCharging() != false)
+    {
+        QTextStream(stdout) << "Voltage: " << batteryData.voltage
+        << " Current: " << batteryData.current
+        << " Total Ah used:  " << batteryStateOfChargeService_.totalAmpHoursUsed()
+        << " Time until fully charged: " << batteryStateOfChargeService_.timeWhenChargedOrDepleted().toString() << endl;
+    }
+    else
+    {
+        QTextStream(stdout) << "Voltage: " << batteryData.voltage
         << " Current: " << batteryData.current
-        << " Total Ah used: " << batteryStateOfChargeService_.totalAmpHoursUsed() << endl;
+        << " Total Ah used:  " << batteryStateOfChargeService_.totalAmpHoursUsed()
+        << " Time until fully depleted: " << batteryStateOfChargeService_.timeWhenChargedOrDepleted().toString() << endl;
 
-    // TODO Print out time till it is depleted or charged.
+    }
```

Comment:
There is duplicated code here, you should move the duplicated code outside of the if else statement.